### ISSUE TITLE
Macos boost configure stage has confusing warnings and errors

### ISSUE THREAD

**Issue by jayvdb** [rose-compiler/rose#169]
I have installed boost with brew on macos, which creates symlinks from `/usr/local`, which I would expect to be findable.

However I get an error
```
../rose/configure
Run configure with --verbose to see more output
Configuring ROSE
Checking Java
Checking GNU Fortran
Checking analyzable languages
cp: ../rose/config/preinclude-cuda.h: No such file or directory
tar: Error opening archive: Failed to open '../rose/cuda-headers.tgz'

Checking message passing interface (MPI)
configure: WARNING: C MPI compiler not found
configure: WARNING: CXX MPI compiler not found
configure: WARNING: F77 MPI compiler not found
configure: WARNING: F MPI compiler not found
Checking GnuCOBOL parse-tree library
Checking modified GNU Octave parser
Checking FLANG library
Checking Python
Checking Boost
--------------------------------------------------------------------------------
configure: error: Unable to find /include/boost/version.hpp . Please specify the right boost installation path with --with-boost=/path/to/boost

For assistance, please visit http://rosecompiler.org or contact the rose-public@nersc.gov mailing list.
```

I then tried the full path
```
../rose/configure --with-boost=/usr/local/Cellar/boost/1.76.0
Run configure with --verbose to see more output
Configuring ROSE
Checking Java
Checking GNU Fortran
Checking analyzable languages
cp: ../rose/config/preinclude-cuda.h: No such file or directory
tar: Error opening archive: Failed to open '../rose/cuda-headers.tgz'

Checking message passing interface (MPI)
configure: WARNING: C MPI compiler not found
configure: WARNING: CXX MPI compiler not found
configure: WARNING: F77 MPI compiler not found
configure: WARNING: F MPI compiler not found
Checking GnuCOBOL parse-tree library
Checking modified GNU Octave parser
Checking FLANG library
Checking Python
Checking Boost
configure: WARNING: detected a previously installed version of boost library: "/usr/local/include/boost"
configure: WARNING: --with-javaport cannot be used with the -isystem option
configure: WARNING: detected previously installed version of boost (please remove older version of Boost before installing ROSE); continuing but expect Boost to be a problem
configure: error: could not link against "boost_system"
```

I then tried with `/usr/local/` with trailing slash
```
../rose/configure --with-boost=/usr/local/
Run configure with --verbose to see more output
Configuring ROSE
Checking Java
Checking GNU Fortran
Checking analyzable languages
cp: ../rose/config/preinclude-cuda.h: No such file or directory
tar: Error opening archive: Failed to open '../rose/cuda-headers.tgz'

Checking message passing interface (MPI)
configure: WARNING: C MPI compiler not found
configure: WARNING: CXX MPI compiler not found
configure: WARNING: F77 MPI compiler not found
configure: WARNING: F MPI compiler not found
Checking GnuCOBOL parse-tree library
Checking modified GNU Octave parser
Checking FLANG library
Checking Python
Checking Boost
configure: WARNING: detected a previously installed version of boost library: "/usr/local/include/boost"
configure: WARNING: --with-javaport cannot be used with the -isystem option
configure: WARNING: detected previously installed version of boost (please remove older version of Boost before installing ROSE); continuing but expect Boost to be a problem
configure: error: could not link against "boost_system"
```

I needed to look at the m4 to finally realise the trailing slash was a problem.
```
../rose/configure --with-boost=/usr/local/Cellar/boost/1.76.0
Run configure with --verbose to see more output
Configuring ROSE
Checking Java
Checking GNU Fortran
Checking analyzable languages
cp: ../rose/config/preinclude-cuda.h: No such file or directory
tar: Error opening archive: Failed to open '../rose/cuda-headers.tgz'

Checking message passing interface (MPI)
configure: WARNING: C MPI compiler not found
configure: WARNING: CXX MPI compiler not found
configure: WARNING: F77 MPI compiler not found
configure: WARNING: F MPI compiler not found
Checking GnuCOBOL parse-tree library
Checking modified GNU Octave parser
Checking FLANG library
Checking Python
Checking Boost
configure: error: could not link against "boost_system"
```

And still, there is a linking error.
```
❯ find /usr/local/lib -name '*boost_system*'
/usr/local/lib/cmake/boost_system-1.76.0
/usr/local/lib/libboost_system-mt.dylib
/usr/local/lib/libboost_system.a
/usr/local/lib/libboost_system-mt.a
/usr/local/lib/libboost_system.dylib
```

Verbose doesnt tell me much more
```
Checking Boost
checking if we should validate your Boost version... yes
checking for boostlib >= 1.51.0... yes
configure: in boost base macro -- Boost version being used is: ""
configure: no previously installed version of boost detected: using boost include directories with normal -I option
configure: rose_boost_version = '107600'
configure: reasonable version of Boost found: 107600
configure: in ROSE/configure: ac_boost_path = "/usr/local"
checking whether the Boost::System library is available... yes
checking for toupper in -lboost_system-mt... no
checking for toupper in -lboost_system-mt... (cached) no
checking for toupper in -lboost_system... no
checking for toupper in -lboost_system... (cached) no
configure: error: could not link against "boost_system"
```

config.log shows the problem
```
g++ -c   -I/usr/local/include conftest.cpp >&5
conftest.cpp:69:1: warning: expression result unused [-Wunused-value]
boost::system::system_category
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
configure:16185: $? = 0
configure:16200: result: yes
configure:16215: checking for toupper in -lboost_system-mt
configure:16240: g++ -o conftest   -I/usr/local/include -Xlinker -rpath /jre/lib/server   -L/usr/local/lib conftest.cpp -lboost_system-mt   >&5
clang: error: no such file or directory: '/jre/lib/server'
configure:16240: $? = 1
```

This is coming from `config/support-java.m4:AM_COND_IF([OS_MACOSX],[LDFLAGS="-Xlinker -rpath ${JAVA_HOME}/jre/lib/server $LDFLAGS"],[])`

`JAVA_HOME` is not set on my system, so it looks like this bit of m4 is causing the clang failure.

However the Java section of configure appears to be working ok

```
Checking Java
checking for Java (javac first, then java, then jvm)... yes
configure: Now verifying aspects of the found java software (java, javac, javah, jar)
checking for java... yes
configure: JAVA = "/usr/bin/../bin/java"
checking JAVA_VERSION...  16.0.1 2021-04-20
checking JAVA_VERSION_MAJOR... 16
checking JAVA_VERSION_MAJOR... 0
checking JAVA_VERSION_PATCH... 1
checking JAVA_VERSION_RELEASE... 2021-04-20
checking for Java JVM include and link options... -I/usr/bin/../include -I/usr/bin/../include/darwin and -L/Library/Java/JavaVirtualMachines/jdk-16.0.1.jdk/Contents/Home/lib/server -ljvm
checking for jar... yes
checking for javac... yes
configure: in support-java: build_os is "darwin20.5.0"
configure: in support-java: LDFLAGS = "-Xlinker -rpath /jre/lib/server "
configure: in support-java: OS_MACOSX = ""
checking jni.h usability... no
checking jni.h presence... no
checking for jni.h... no
ROSE can't find the jni.h in a default directory (this is good since it will be included correctly).
configure: summary of Java information:
configure:     The --with-java switch specified:       try
configure:     Installation path (JAVA_PATH):          /usr/bin/..
configure:     C++ header switches (JAVA_JVM_INCLUDE): -I/usr/bin/../include -I/usr/bin/../include/darwin
configure:     C++ link switches (JAVA_JVM_LINK):      -L/Library/Java/JavaVirtualMachines/jdk-16.0.1.jdk/Contents/Home/lib/server -ljvm
configure:     Runtime command (JAVA):                 /usr/bin/../bin/java
configure:     Compiler command (JAVAC):               /usr/bin/../bin/javac
configure:     Archive tool (JAR):                     /usr/bin/../bin/jar
```

I see some `JAVA_HOME` debugging stuff in `config/support-rose.m4`.  fwiw, I dont have a `javaconfig` on BigSur.

brew `java` is currently an alias for [https://formulae.brew.sh/formula/openjdk](`openjdk`), which installs into e.g. `/usr/local/Cellar/openjdk/15.0.2/`, and would be my preference to use.

However if I try `JAVA_HOME=/usr/local/Cellar/openjdk/15.0.2/ ../rose/configure ..`, the Java section appears to work correctly, but then boost still fails.

```
Checking Java
checking for Java (javac first, then java, then jvm)... yes
configure: Now verifying aspects of the found java software (java, javac, javah, jar)
checking for java... yes
configure: JAVA = "/usr/local/Cellar/openjdk/15.0.2//bin/java"
checking JAVA_VERSION...  15.0.2 2021-01-19
checking JAVA_VERSION_MAJOR... 15
checking JAVA_VERSION_MAJOR... 0
checking JAVA_VERSION_PATCH... 2
checking JAVA_VERSION_RELEASE... 2021-01-19
checking for Java JVM include and link options... -I/usr/local/Cellar/openjdk/15.0.2//include -I/usr/local/Cellar/openjdk/15.0.2//include/darwin and -L/usr/local/Cellar/openjdk/15.0.2/libexec/openjdk.jdk/Contents/Home/lib/server -ljvm
checking for jar... yes
checking for javac... yes
configure: in support-java: build_os is "darwin20.5.0"
configure: in support-java: LDFLAGS = "-Xlinker -rpath /usr/local/Cellar/openjdk/15.0.2//jre/lib/server "
configure: in support-java: OS_MACOSX = ""
checking jni.h usability... no
checking jni.h presence... no
checking for jni.h... no
ROSE can't find the jni.h in a default directory (this is good since it will be included correctly).
configure: summary of Java information:
configure:     The --with-java switch specified:       try
configure:     Installation path (JAVA_PATH):          /usr/local/Cellar/openjdk/15.0.2/
configure:     C++ header switches (JAVA_JVM_INCLUDE): -I/usr/local/Cellar/openjdk/15.0.2//include -I/usr/local/Cellar/openjdk/15.0.2//include/darwin
configure:     C++ link switches (JAVA_JVM_LINK):      -L/usr/local/Cellar/openjdk/15.0.2/libexec/openjdk.jdk/Contents/Home/lib/server -ljvm
configure:     Runtime command (JAVA):                 /usr/local/Cellar/openjdk/15.0.2//bin/java
configure:     Compiler command (JAVAC):               /usr/local/Cellar/openjdk/15.0.2//bin/javac
configure:     Archive tool (JAR):                     /usr/local/Cellar/openjdk/15.0.2//bin/jar
```

config.log shows
`clang: error: no such file or directory: '/usr/local/Cellar/openjdk/15.0.2/jre/lib/server'`

The Java section correctly found the libs, but they are not at `${JAVA_HOME}/jre/lib/server` causing the problem.
```
ls /usr/local/Cellar/openjdk/15.0.2/libexec/openjdk.jdk/Contents/Home/lib/server/
classes.jsa
classes_nocoops.jsa
libjsig.dylib
libjvm.dylib
```

If I go to `/usr/local/Cellar/openjdk/15.0.2/`, and create `jre` and symlink `lib` to `/usr/local/Cellar/openjdk/15.0.2/libexec/openjdk.jdk/Contents/Home/lib/` , Boost now is ok.

**Comment by jayvdb** [comment]
I then hit `/usr/bin/bison` not being accepted, and `brew install bison` is keg-only, so it needs to be found in
/usr/local/Cellar/bison/3.7.6/bin/bison

And then from `config/choose-backend-compiler.m4`, rather unfriendly error
```
Checking ROSE backend
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
configure: error: in `/Users/john.vandenberg/transpilers/rose-build':
configure: error: unknown or unsupported version of XCode: XCODE_VERSION_MINOR = "0"
```
I have 12.0 according to the configure
```
configure:19108: (g++ but really clang) C++ back-end compiler major version number = "12"
configure:19110: (g++ but really clang) C++ back-end compiler minor version number = "0"
```
This error was added in https://github.com/rose-compiler/rose/commit/fd0b7dff47fbbbe9bdd4d102cdde1fd3fa10a7b2#diff-08588112b6cc652eddf8eeb1ca692658029f6e6e0bad7aa36cf24e408a0fa9e7R313

**Comment by jayvdb** [comment]
After patching `config/choose-backend-compiler.m4` and `config/compiler-defs.m4` to accept llvm 12.0, I encounter
https://github.com/rose-compiler/rose/issues/170 , and a lot of other oddities in the configure which I'll dump here rather than create lots of separate issues about
```
Checking ROSE backend
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
find: /usr/include: No such file or directory
find: /usr/include: No such file or directory
sed: -e expression #1, char 1: unknown command: `.'
Checking frontend compiler
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Checking backend compiler
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Checking backend C/C++ compiler specific references
Checking Valgrind
Checking ATERM library
Checking STRATEGO library
Checking LLVM
../rose/configure: line 34331: echo: write error: Broken pipe
../rose/configure: line 34332: echo: write error: Broken pipe
Checking Doxygen
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Checking system capabilities
Checking binary analysis
ld: warning: The i386 architecture is deprecated for macOS (remove from the Xcode build setting: ARCHS)
ld: warning: ignoring file /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd, missing required architecture i386 in file /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd (3 slices)
ld: dynamic main executables must link with libSystem.dylib for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Configuration summary:
  Compilers for ROSE source code
    C++ compiler                     g++
    C++ vendor                       llvm
    C++ version                      12.0.5
    C++ language                     gnu++03
    Bison version                    3.7.6
  Languages supported for analysis
    Binaries                         yes
    C                                yes
    C++                              yes
    C preprocessor                   yes
    Cuda                             yes
    Fortran                          yes
    Java                             yes
    OpenCL                           yes
    PHP                              yes
    Python                           no
  Boost library
    location                         /usr/local
    version constant                 107600
    asio library                     none
    chrono library                   -lboost_chrono-mt
    date/time library                -lboost_date_time-mt
    filesystem library               -lboost_filesystem-mt
    iostreams library                -lboost_iostreams-mt
    program options library          -lboost_program_options-mt
    python library                   none
    random library                   -lboost_random-mt
    regex library                    -lboost_regex-mt
    serialization library            -lboost_serialization-mt
    signals library                  none
    system library                   -lboost_system-mt
    test exec monitor library        none
    thread library                   -lboost_thread-mt
    unit test framework library      none
    wave library                     -lboost_wave-mt
    wserialization library           none
  Binary analysis support
    Capstone          (ARM decoding) none
    Dlib          (graph algorithms) none
    DWARF     (ELF format debugging) none
    ELF       (only needed by DWARF) none
    i386 support                     no
    magic      (file identification) none
    readline     (interactive input) none
    SQLite3     (database in a file) none
    YAML-cpp         (configuration) none
    yices executable    (SMT solver) none
    yices library                    none
    z3 version          (SMT solver) unknown
    z3 executable                    none
    z3 library                       none
  C/C++ analysis support
    frontend C++ parser              EDG-5.0
    compiling EDG from source        no
    generating EDG binary tarball    no
    backend C++ compiler             g++
    C++ vendor                       clang
    C++ version                      10.0.0
    backend C compiler               gcc
  Fortran analysis support
    fortran compiler                 gfortran
  Java analysis support
    installation path            	   /usr/local/Cellar/openjdk/15.0.2/
    C++ header switches          	   -I/usr/local/Cellar/openjdk/15.0.2//include -I/usr/local/Cellar/openjdk/15.0.2//include/darwin
    C++ link switches            	   -L/usr/local/Cellar/openjdk/15.0.2/libexec/openjdk.jdk/Contents/Home/lib/server -ljvm
    runtime command                  /usr/local/Cellar/openjdk/15.0.2//bin/java
    compiler command                 /usr/local/Cellar/openjdk/15.0.2//bin/javac
    archive tool                     /usr/local/Cellar/openjdk/15.0.2//bin/jar
  Doxygen documentation generator
    executable                       doxygen
    version                          1.9.1
    indexer                          none
    stubs for undocumented entities  NO
    document private entities        NO
    document static entities         NO
    document internal entities       NO
    hide undocumented members        NO
    hide undocumented classes        NO
    hide friend compounds            YES
Checking blacklisted configurations
Generating files
Makefile:2404: warning: overriding commands for target `check-recursive'
Makefile:1889: warning: ignoring old commands for target `check-recursive'
Makefile:3352: warning: overriding commands for target `librose_lcov_web_pages'
Makefile:3278: warning: ignoring old commands for target `librose_lcov_web_pages'
Makefile:2404: warning: overriding commands for target `check-recursive'
Makefile:1889: warning: ignoring old commands for target `check-recursive'
Makefile:3352: warning: overriding commands for target `librose_lcov_web_pages'
Makefile:3278: warning: ignoring old commands for target `librose_lcov_web_pages'
(CDPATH="${ZSH_VERSION+.}:" && cd ../rose && /bin/sh '/Users/john.vandenberg/transpilers/rose/config/missing' autoheader)
configure.ac:139: warning: AC_REQUIRE: `_LT_COMPILER_PIC' was expanded before it was required
configure.ac:139: http://www.gnu.org/software/autoconf/manual/autoconf.html#Expanded-Before-Required
config/ltdl.m4:904: LT_FUNC_DLSYM_USCORE is expanded from...
config/ltdl.m4:421: _LTDL_SETUP is expanded from...
config/ltdl.m4:349: LTDL_INIT is expanded from...
config/ltdl.m4:352: AC_LIB_LTDL is expanded from...
config/support-rose.m4:1756: ROSE_SUPPORT_ROSE_PART_2 is expanded from...
configure.ac:139: the top level
configure.ac:141: warning: AC_LANG_CONFTEST: no AC_LANG_SOURCE call detected in body
../../lib/autoconf/lang.m4:197: AC_LANG_CONFTEST is expanded from...
../../lib/autoconf/general.m4:2598: _AC_COMPILE_IFELSE is expanded from...
../../lib/autoconf/general.m4:2608: AC_COMPILE_IFELSE is expanded from...
../../lib/m4sugar/m4sh.m4:643: AS_IF is expanded from...
../../lib/autoconf/general.m4:2046: AC_CACHE_VAL is expanded from...
../../lib/autoconf/general.m4:2059: AC_CACHE_CHECK is expanded from...
config/ax_boost_serialization.m4:140: AX_BOOST_SERIALIZATION is expanded from...
config/support-binary.m4:176: ROSE_SUPPORT_BINARY is expanded from...
config/support-rose.m4:1952: ROSE_SUPPORT_ROSE_PART_4 is expanded from...
configure.ac:141: the top level
rm -f stamp-h1
touch ../rose/rose_config.h.in
cd . && /bin/sh ./config.status rose_config.h
config.status: creating rose_config.h
config.status: rose_config.h is unchanged
  GEN     rosePublicConfig.h
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
appending configuration tag "CXX" to libtool
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
appending configuration tag "F77" to libtool
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory

+--------------------------------+
| Configure Terminated Normally! |
+--------------------------------+

Now type 'make core -jN' and then 'make install-core -jN' to only build essential libraries and tools (Recommended).
Type 'make -jN' and 'make install -jN' to build and install everything (takes a very long time).
N is the number of concurrent processes you want to use for your build.
Thank you for using ROSE.
```

**Comment by jayvdb** [comment]
And not surprising, boost link errors:
```
  ...
  CXX      libroseSawyer_la-Synchronization.lo
../../../../rose/src/util/Sawyer/Synchronization.C:31:4: warning: "Sawyer multi-threading is disabled; even documented thread-safe functions are unsafe!" [-W#warnings]
#  warning "Sawyer multi-threading is disabled; even documented thread-safe functions are unsafe!"
   ^
1 warning generated.
  CXXLD    libroseSawyer.la
  CXXLD    stringify-enums
Undefined symbols for architecture x86_64:
  "boost::scoped_static_mutex_lock::scoped_static_mutex_lock(boost::static_mutex&, bool)", referenced from:
      boost::object_cache<boost::re_detail_500::cpp_regex_traits_base<char>, boost::re_detail_500::cpp_regex_traits_implementation<char> >::get(boost::re_detail_500::cpp_regex_traits_base<char> const&, unsigned long) in stringifyEnums.o
      boost::cpp_regex_traits<char>::get_catalog_name() in stringifyEnums.o
      boost::re_detail_500::get_mem_block() in stringifyEnums.o
      boost::re_detail_500::perl_matcher<std::__1::__wrap_iter<char const*>, std::__1::allocator<boost::sub_match<std::__1::__wrap_iter<char const*> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::unwind_extra_block(bool) in stringifyEnums.o
      boost::re_detail_500::save_state_init::~save_state_init() in stringifyEnums.o
      boost::re_detail_500::get_mem_block() in libroseSawyer.a(libroseSawyer_la-CommandLine.o)
      boost::re_detail_500::perl_matcher<std::__1::__wrap_iter<char const*>, std::__1::allocator<boost::sub_match<std::__1::__wrap_iter<char const*> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::unwind_extra_block(bool) in libroseSawyer.a(libroseSawyer_la-CommandLine.o)
      ...
  "boost::scoped_static_mutex_lock::~scoped_static_mutex_lock()", referenced from:
      boost::object_cache<boost::re_detail_500::cpp_regex_traits_base<char>, boost::re_detail_500::cpp_regex_traits_implementation<char> >::get(boost::re_detail_500::cpp_regex_traits_base<char> const&, unsigned long) in stringifyEnums.o
      boost::cpp_regex_traits<char>::get_catalog_name() in stringifyEnums.o
      boost::re_detail_500::get_mem_block() in stringifyEnums.o
      boost::re_detail_500::perl_matcher<std::__1::__wrap_iter<char const*>, std::__1::allocator<boost::sub_match<std::__1::__wrap_iter<char const*> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::unwind_extra_block(bool) in stringifyEnums.o
      boost::re_detail_500::save_state_init::~save_state_init() in stringifyEnums.o
      boost::re_detail_500::get_mem_block() in libroseSawyer.a(libroseSawyer_la-CommandLine.o)
      boost::re_detail_500::perl_matcher<std::__1::__wrap_iter<char const*>, std::__1::allocator<boost::sub_match<std::__1::__wrap_iter<char const*> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::unwind_extra_block(bool) in libroseSawyer.a(libroseSawyer_la-CommandLine.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [stringify-enums] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [core] Error 2
```

**Comment by peihunglin** [comment]
For you boost issue, can you try to use clang and clang++ to build ROSE?  You can add "CXXFLAGS=-std=c++14 CC=clang CXX=clang++" to you configure command.
For JAVA_HOME, we currently use "/usr/libexec/java_home" with filters options to setup the environment variable.

**Comment by jayvdb** [comment]
I am using clang to build ROSE.
Using that CXXFLAGS with v0.11.34.0 and v0.11.35.0
```
Making all in InstructionSemantics2
  CXX      BaseSemantics.lo
  CXX      BaseSemanticsDispatcher.lo
  CXX      BaseSemanticsException.lo
  CXX      BaseSemanticsMemoryState.lo
In file included from ../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsDispatcher.C:4:
../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsDispatcher.h:42:26: error: implicit instantiation of undefined template 'boost::enable_shared_from_this<Rose::BinaryAnalysis::InstructionSemantics2::BaseSemantics::Dispatcher>'
class Dispatcher: public boost::enable_shared_from_this<Dispatcher> {
                         ^
/usr/local/include/boost/smart_ptr/shared_ptr.hpp:56:25: note: template is declared here
template<class T> class enable_shared_from_this;
                        ^
In file included from ../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsDispatcher.C:6:
../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsRiscOperators.h:46:29: error: implicit instantiation of undefined template 'boost::enable_shared_from_this<Rose::BinaryAnalysis::InstructionSemantics2::BaseSemantics::RiscOperators>'
class RiscOperators: public boost::enable_shared_from_this<RiscOperators> {
                            ^
/usr/local/include/boost/smart_ptr/shared_ptr.hpp:56:25: note: template is declared here
template<class T> class enable_shared_from_this;
                        ^
In file included from ../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsDispatcher.C:6:
../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsRiscOperators.h:249:70: error: use of undeclared identifier 'shared_from_this'
    WithFormatter with_format(Formatter &fmt) { return WithFormatter(shared_from_this(), fmt); }
                                                                     ^
In file included from ../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsDispatcher.C:7:
../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsState.h:37:21: error: implicit instantiation of undefined template 'boost::enable_shared_from_this<Rose::BinaryAnalysis::InstructionSemantics2::BaseSemantics::State>'
class State: public boost::enable_shared_from_this<State> {
                    ^
/usr/local/include/boost/smart_ptr/shared_ptr.hpp:56:25: note: template is declared here
template<class T> class enable_shared_from_this;
                        ^
In file included from ../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsDispatcher.C:7:
../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsState.h:99:58: error: use of undeclared identifier 'shared_from_this'
        StatePtr self = boost::const_pointer_cast<State>(shared_from_this());
                                                         ^
../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsState.h:252:70: error: use of undeclared identifier 'shared_from_this'
    WithFormatter with_format(Formatter &fmt) { return WithFormatter(shared_from_this(), fmt); }
                                                                     ^
../../../../../rose/src/Rose/BinaryAnalysis/InstructionSemantics2/BaseSemanticsDispatcher.C:90:24: error: use of undeclared identifier 'shared_from_this'
        iproc->process(shared_from_this(), insn);
                       ^
7 errors generated.
make[4]: *** [BaseSemanticsDispatcher.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
  CXX      BaseSemanticsMerger.lo
  CXX      BaseSemanticsRegisterState.lo
make[3]: *** [all-recursive] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [core] Error 2
```

**Comment by matzke1** [comment]
The compile errors related to `boost::enable_shared_from_this` should be fixed now. They were due to missing #include <boost/enable_shared_from_this.hpp>, fixed by commit 6cfeea07bea4f5a668de602ac4a5d9044e1438a2 on June 2nd.
